packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts

```diff
@@ -218,4 +226,36 @@ export class MetricsExplorerService<
             fields: allFields,
         };
     }
+
+    /**
+     * Helper method to determine the correct time dimension configuration
+     * It covers the case where the metric has a default time dimension
+     * and the case where the metric has available time dimensions (could be multiple and from different tables that are joined)
+     */
+    private static getTimeDimensionConfig(
```

Comment:
I was thinking of moving this to the catalogService but maybe not - seems tied to this more. 
Would a util for this, and the other 2 utils in CatalogService be a better placement? As in, a separate file?

Reply:
this can be inside metric explorer utils (common) and maybe reused in both catalog and here?

Reply:
[refactor: move utils to metricsExplorer util](https://github.com/lightdash/lightdash/pull/12701/commits/ded6c6b9d70703fa878a7127d1714802016f3a42)